test/onnx/internal/fx/test_frontend_consistency.py

```diff
@@ -0,0 +1,294 @@
+MODULES_DB = [
+    common_modules.ModuleInfo(MLPModel, module_inputs_func=module_inputs_mlp_model),
+    common_modules.ModuleInfo(
+        SingleAddModel, module_inputs_func=module_inputs_single_add
+    ),
+]
+
+# END OF SECTION TO MODIFY to add or modify nn.Module test cases ###############
+
+
+class _TestFxFrontendConsistency(common_utils.TestCase):
+    frontend_info: FrontendInfo
+
+    @common_modules.modules(
+        MODULES_DB, train_eval_mode=common_modules.TrainEvalMode.eval_only
+    )
+    def test_output_match(
+        self,
+        device: str,
+        dtype: torch.dtype,
+        module_info: common_modules.ModuleInfo,
+        training: bool,
+    ):
+        """Base test method for testing each module_info with fx frontend."""
+        # device is provided by instantiate_device_type_tests, but we only want to run in cpu.
+        assert device == "cpu"
+
+        if skip_reason := self.frontend_info.skips.get(module_info.formatted_name):
+            pytest.skip(skip_reason)
+
+        fx_frontend = self.frontend_info.fx_frontend
+
+        module_cls = module_info.module_cls
+        module_inputs = module_info.module_inputs_func(
+            module_info,
+            device=device,
+            dtype=dtype,
+            requires_grad=False,
+            training=training,
+        )
+
+        for (i, module_input) in enumerate(module_inputs):
+            with self.subTest(
+                module_name=module_info.formatted_name,
+                sample_num=i,
+                fx_frontend=fx_frontend.name,
+            ):
+                init_args, init_kwargs = (
+                    module_input.constructor_input.args,
+                    module_input.constructor_input.kwargs,
+                )
+                model = module_cls(*init_args, **init_kwargs)
+                model.to(device).to(dtype)
+                model.train(training)
+
+                args, kwargs = (
+                    module_input.forward_input.args,
+                    module_input.forward_input.kwargs,
+                )
+
+                if isinstance(fx_frontend, frontend.FxFrontendUnpackKwargs):
+                    graph_module, new_args = fx_frontend.trace(model, *args, **kwargs)
+                    fx_outputs = graph_module(*new_args)
+                else:
+                    graph_module = fx_frontend.trace(model, *args, **kwargs)
+                    fx_outputs = graph_module(*args, **kwargs)
+
+                outputs = model(*args, **kwargs)
+
+                self.assertEqual(
+                    pytree.tree_flatten(outputs)[0],
+                    pytree.tree_flatten(fx_outputs)[0],
+                )
+
+
+# 'Parameterize' the test class to run for each fx frontend.
+# Adopted from `parameterized` package.
+# The reason not using `parameterized` package is that it doesn't support
+# applying additional `common_device_type.instantiate_device_type_tests`.
+for frontend_info in FX_FRONTEND_INFOS:
```

Comment:
I am still not convinced to use code-gen style way to generate tests. We can discuss offline.

Reply:
If the test body is common like we have now, I think it's worth parameterizing. Alternatively we _could_ extract the test body into a function, create multiple modules that call this function etc, but that would mean either (1) two growing list to maintain (FrontendInfo and the test classes) or (2) scattered config (no FronendInfo, skips in every method)